pkgs/data/fonts/nerdfonts/default.nix

```diff
@@ -1,26 +1,50 @@
-{ stdenv, fetchFromGitHub, which, withFont ? "" }:
+{ lib, stdenv, fetchFromGitHub
+, which, fonts ? []
+, patchFontDrvs ? [], patchFlags ? [ "mono" "windows" "complete" "careful" ], fontforge ? null }:
+
+assert patchFontDrvs != [] -> fontforge != null;
 
 stdenv.mkDerivation rec {
-  version = "2.1.0";
   pname = "nerdfonts";
+  version = "2.1.0";
+
   src = fetchFromGitHub {
     owner = "ryanoasis";
     repo = "nerd-fonts";
     rev = version;
     sha256 = "1la79y16k9rwcl2zsxk73c0kgdms2ma43kpjfqnq5jlbfdj0niwg";
   };
-  dontPatchShebangs = true;
-  buildInputs = [ which ];
+
+  buildInputs = [ which fontforge ];
+
   patchPhase = ''
     patchShebangs install.sh
     sed -i -e 's|font_dir="\$HOME/.local/share/fonts|font_dir="$out/share/fonts/truetype|g' install.sh
   '';
+
+  ffPatcher = "fontforge -script font-patcher ${lib.concatMapStringsSep " " (flag: "--${flag}") patchFlags}";
+
   installPhase = ''
+    mkdir -p $out/share/fonts/opentype
     mkdir -p $out/share/fonts/truetype
-    ./install.sh ${withFont}
+
+    ${lib.concatMapStringsSep "\n" (drv: ''
+      find ${drv.out}/share/fonts -name '*.otf' \
+        -exec $ffPatcher --outputdir $out/share/fonts/opentype --extension otf "{}" \;
+      find ${drv.out}/share/fonts -name '*.ttf' \
+        -exec $ffPatcher --outputdir $out/share/fonts/truetype --extension ttf "{}" \;
+    '') patchFontDrvs}
+
+    ${lib.optionalString (fonts != null) ''
+      ./install.sh ${lib.concatStringsSep " " fonts}
+
+      find $out/share/fonts -name '*.otf' -exec mv "{}" $out/share/fonts/opentype \;
```

Comment:
Is it not possible to do `install -D *.otf -t $out/share/fonts/opentype`?
This would also shave off the `mkdir -p`

Reply:
Done.

Reply:
Errors out if no `*.otf` fonts exists, reverted.